src/assets/CollectiblesController.ts

```diff
@@ -236,14 +237,27 @@ export class CollectiblesController extends BaseController<
   ): Promise<CollectibleMetadata> {
     const tokenURI = this.getCollectibleApi(contractAddress, tokenId);
     let collectibleInformation: ApiCollectible;
-
     /* istanbul ignore if */
-    if (this.openSeaApiKey) {
-      collectibleInformation = await handleFetch(tokenURI, {
-        headers: { 'X-API-KEY': this.openSeaApiKey },
-      });
-    } else {
+    try {
       collectibleInformation = await handleFetch(tokenURI);
+    } catch (e) {
+      logOrRethrowError(e);
+
+      if (this.openSeaApiKey) {
+        collectibleInformation = await handleFetch(
```

Comment:
The "log and re-throw" pattern should be applied here as well. Otherwise the intended defaults will not be applied if both calls fail.

Maybe this "log and throw" step can be moved a function that is called in place of `handleFetch`?

Reply:
@Gudahtt per my comment below I have sketched out two patterns to handle errors. Not sure which I prefer. When you have a moment could you provide input?